src/modules/builtin/mv.rs

```diff
@@ -0,0 +1,61 @@
+use std::mem::swap;
+
+use heraclitus_compiler::prelude::*;
+use crate::modules::expression::expr::Expr;
+use crate::modules::condition::failed::Failed;
+use crate::translate::module::TranslateModule;
+use crate::docs::module::DocumentationModule;
+use crate::utils::{ParserMetadata, TranslateMetadata};
+use crate::modules::command::modifier::CommandModifier;
+
+#[derive(Debug, Clone)]
+pub struct Mv {
+    source: Expr,
+    destination: Expr,
+    modifier: CommandModifier,
+    failed: Failed,
+}
+
+impl SyntaxModule<ParserMetadata> for Mv {
+    syntax_name!("MoveFiles");
+
+    fn new() -> Self {
+        Mv {
+            source: Expr::new(),
+            destination: Expr::new(),
+            failed: Failed::new(),
+            modifier: CommandModifier::new().parse_expr(),
+        }
+    }
+
+    fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
+        syntax(meta, &mut self.modifier)?;
+        self.modifier.use_modifiers(meta, |_this, meta| {
+            token(meta, "mv")?;
+            syntax(meta, &mut self.source)?;
+            syntax(meta, &mut self.destination)?;
+            match syntax(meta, &mut self.failed) {
+                Ok(_) => Ok(()),
+                Err(err) => Err(err)
+            }
+        })
+    }
+}
+
+impl TranslateModule for Mv {
+    fn translate(&self, meta: &mut TranslateMetadata) -> String {
+        let source = self.source.translate(meta);
+        let destination = self.destination.translate(meta);
+        let failed = self.failed.translate(meta);
+        let mut is_silent = self.modifier.is_silent || meta.silenced;
+        swap(&mut is_silent, &mut meta.silenced);
+        let silent = meta.gen_silent();
+        format!("mv {source} {destination}{silent}\n{failed}").trim_end().to_string()
```

Comment:
Also we probably don't need to handle silent in bash since `mv` outputs nothing. Right?

Reply:
I would say it might has a usecase as `mv` echos an error when it fails so you might use it like that to ignore the error message
```
silent mv "file_that_doesnt_exists" "another_file" failed {
    // handing the error
}
```

Reply:
Oh I understand. Then why don't we swap back the silent value?

```suggestion
        let mut is_silent = self.modifier.is_silent || meta.silenced;
        // Swap modifier
        swap(&mut is_silent, &mut meta.silenced);
        let silent = meta.gen_silent();
        // Swap back
        swap(&mut is_silent, &mut meta.silenced);
        format!("mv {source} {destination}{silent}\n{failed}").trim_end().to_string()
```

This should be also implemented as `use_modifier_context` in translation layer - just as it is in parser. We can create an issue for that.

Reply:
Right, totally forget to swap them back.

Reply:
Updated the pr.